cns/logger/log.go

```diff
@@ -22,113 +24,90 @@ var (
 type CNSLogger struct {
 	logger               *log.Logger
 	th                   aitelemetry.TelemetryHandle
-	Orchestrator         string
-	NodeID               string
 	DisableTraceLogging  bool
 	DisableMetricLogging bool
 	DisableEventLogging  bool
-}
 
-// Initialize CNS Logger
-func InitLogger(fileName string, logLevel, logTarget int, logDir string) {
-	Log = &CNSLogger{
-		logger: log.NewLogger(fileName, logLevel, logTarget, logDir),
-	}
+	m            sync.RWMutex
+	Orchestrator string
+	NodeID       string
```

Comment:
motivation for changes in this file are making this cns logger easier to stub out in tests, as well as addressing what I'm pretty sure is a race condition when setting these fields, since the logging methods here read them all the time but they are set, possibly multiple times, from an http handler.